core/config/config_test.go

```diff
@@ -180,6 +180,19 @@ func TestSimpleConfigValues(t *testing.T) {
 	assert.NoError(t, v.PopulateStruct(nested))
 }
 
+func TestGetAsIntegerValue(t *testing.T) {
+	cv := NewValue(NewStaticProvider(nil), "key", float32(2), true, Integer, nil)
```

Comment:
https://blog.golang.org/subtests

basically build up a set of test cases and range over them